### PR TITLE
Squash bug in `SoilEvaporation`

### LINKED ISSUE #841: Evaporation from the soil profile is buggy
Water evaporation from the soil profile should peak around ~3 mm and cycle consistently over years.

## Expected Behavior

Water should evaporate out of the soil at a rate of ~3 mm and should consistently change throughout the seasons.
desired_soil_evaporation

## Current Behavior

Evaporation from the soil profile peaks at about 1/10 if what it should be peaking at. It also bottoms out intermittently while it should be peaking.
buggy_rufas_evaporation

## Steps to Reproduce

1. Run `barnyard_metadata.json`
2. Observe results.
3.
4.
5.

### PR DESCRIPTION
**Pull request by Isgotkowitz**
This PR fixes a bug in the equation that calculates the potential evapotranspiration from a Field, more specifically a value was -0.5 instead of 0.5.

I also took this opportunity to pay off some tech debt related to evapotranspiration. This includes updating docstrings, adjusting method signatures, eliminating duplicate or unused attributes, recording values with better resolution, and rewriting methods/tests to be cleaner and follow best practices.

Closes #841

Both graphs below plot total evaporation from the soil profile on a daily basis, from day 244 of 2009 to day 243 of 2016.
Before:
buggy_rufas_evaporation

After:
first_fixed_rufas_evaporation

### CODE REVIEW

File: RUFAS/routines/field/field/field.py [hunk 1]
```diff
@@ -1200,20 +1201,15 @@ def _determine_potential_evapotranspiration(extra_terrestrial_radiation: float,
...
-                                            extra_terrestrial_radiation * ((max_air_temp - min_air_temp) ** (-0.5))
-                                            * (calculated_avg_air_temp + 17.8)) / latent_heat_vaporization
-            return max(0.0, potential_evapotranspiration)
-        else:
-            latent_heat_vaporization = Field._determine_latent_heat_vaporization(avg_air_temp)
-            potential_evapotranspiration = (0.0023 *
-                                            extra_terrestrial_radiation * ((max_air_temp - min_air_temp) ** (-0.5))
-                                            * (avg_air_temp + 17.8)) / latent_heat_vaporization
-            return max(0.0, potential_evapotranspiration)
+        avg_air_temp = avg_air_temp if avg_air_temp else (max_air_temp + min_air_temp) / 2
```
Comment (JoeWaddell) [inline RUFAS/routines/field/field/field.py:1208]: Great seeing a both big reduction in the number of lines and readability.

**Review by JoeWaddell — APPROVED** [review]
LGTM: elegant fixes and a nice job ensuring full test coverage!

**Review by allisterakun — APPROVED** [review]
LGTM

**Comment by Isgotkowitz** [comment]
@allisterakun @JoeWaddell thanks for the reviews:)

**Pull request merged**